plugins/es-repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureStorageSettings.java

```diff
@@ -176,6 +179,24 @@ private String buildConnectString(String account,
         return connectionStringBuilder.toString();
     }
 
+    /**
+     * Verify that the endpoint is a valid URI with a nonnull scheme and host.
+     * Otherwise, the Azure SDK will throw an NPE exception if host is null.
+     */
+    private void validateEndpoint(String endpoint, String endpointName) {
+        if (Strings.isNullOrEmpty(endpoint)) {
+            return;
+        }
```

Comment:
Do I get it right that Azure SDK doesn't accept nulls in URI parts (host/scheme) but the whole URI can be null?

Reply:
Yes